File: Projects/Project_05a_1D_Elliptic/For_Students/julia/exact_solution.jl
#---------------------------------------------------------------------#
#This function computes the Initial & Exact Solutions.
#Written by F.X. Giraldo on April 19, 2019
#           Department of Applied Mathematics
#           Naval Postgraduate School
#           Monterey; CA 93943-5216
#---------------------------------------------------------------------#
function exact_solution(coord,Npoin,case,DFloat)

    #Set some constants
    c=2*π

    #Initialize
    qe=zeros(DFloat,Npoin)
    qeₓ=zeros(DFloat,Npoin)
    fe=zeros(DFloat,Npoin)

    #Generate Grid Points
    for I=1:Npoin
        x=coord[I]
        if (case == 1) #Gaussian
            qe[I]=sin(c*x)
            qeₓ[I]=-c*cos(c*x)
            fe[I]=-c^2*sin(c*x)
        end
    end #I
    return (qe,qeₓ,fe)
end

File: Projects/Project_05a_1D_Elliptic/For_Students/matlab/exact_solution.m
%---------------------------------------------------------------------%
%This function computes the Initial and Analytic Solutions.
%Written by F.X. Giraldo on 10/2003
%           Department of Applied Mathematics
%           Naval Postgraduate School 
%           Monterey, CA 93943-5216
%---------------------------------------------------------------------%
function [qe,qe_x,fe] = exact_solution(coord,npoin,icase)

%Initialize
qe=zeros(npoin,1);
qe_x=zeros(npoin,1);
fe=zeros(npoin,1);
c=pi*2;

%Generate Grid Points
for ip=1:npoin
   x=coord(ip);
   if (icase == 1)
        qe(ip)=sin(c*x);
        qe_x(ip)=-c*cos(c*x);
        fe(ip)=-c^2*sin(c*x);
   end
end %ip      


      
